flink-metrics/flink-metrics-core/src/main/java/org/apache/flink/metrics/MetricGroup.java

```diff
@@ -87,9 +89,27 @@
 	 * @param gauge gauge to register
 	 * @param <T>   return type of the gauge
 	 * @return the given gauge
+	 * @deprecated use {@link #register(String, NumberGauge)} or {@link #register(String, StringGauge)} instead
 	 */
+	@Deprecated
 	<T, G extends Gauge<T>> G gauge(String name, G gauge);
 
+	/**
+	 * Registers a new {@link org.apache.flink.metrics.NumberGauge} with Flink.
```

Comment:
Minor: You don't need the fully qualified class name, the JavaDocs actually read easier with just the class names.